### PR TITLE
refactor: proxy wiring for EventSignatureValidator

### LINKED ISSUE #12133: Use proxy wiring for `EventSignatureValidator`
- use the new proxy wiring for the `EventSignatureValidator`
- Extract a `EventSignatureValidator` interface to be compatible

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
This PR creates an interface and component wiring for the `EventSignatureValidator`. 

**Related issue(s)**:

Fixes #12133

### CODE REVIEW

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java [hunk 1]
```diff
@@ -16,248 +16,37 @@
...
-                    EXCEPTION.getMarker(),
-                    "Event failed signature check. Event: {}, Signature: {}, Hash: {}",
-                    event,
-                    CommonUtils.hex(event.getUnhashedData().getSignature()),
-                    event.getHashedData().getHash());
-        }
-
-        return isSignatureValid;
-    }
+public interface EventSignatureValidator {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java:25]: Missing a doc for the interface.

The pattern I've been following is to transfer the doc from the original class into the interface, and then give the impl class a doc that just says it's the default implementation of the interface.

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java:28]: thanks for the catch!
updated

**Review by alittley — APPROVED** [review]

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — COMMENTED** [review]
Can you post the updated wiring diagram?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java [hunk 2]
```diff
@@ -16,248 +16,40 @@
...
 
     /**
      * Set the non-ancient event window that defines the minimum threshold required for an event to be non-ancient
      *
      * @param nonAncientEventWindow the non-ancient event window
      */
-    public void setNonAncientEventWindow(@NonNull final NonAncientEventWindow nonAncientEventWindow) {
-        this.nonAncientEventWindow = Objects.requireNonNull(nonAncientEventWindow);
-    }
+    @InputWireLabel("NonAncientEventWindow")
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java:45]: This is different than the original. Not necessarily a problem, but want to see the wiring diagram to make sure it still looks ok.

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java:45]: wiring diagram added.

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java:45]: removing wiring diagram as it creates a merge conflict.

**Review by alittley — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Pull request merged**